src/pages/academy/adminPanel/subcomponents/NotificationConfigPanel.tsx

```diff
@@ -0,0 +1,223 @@
+   * value of assessmentConfig.
+   *
+   * Also, useState causes a flicker in ag-grid during rerenders. Thus we use this mutable ref and
+   * ag-grid's API to update cell values instead.
+   */
+  const notificationConfig = React.useRef(session.notificationConfigs) as React.MutableRefObject<
+    NotificationConfiguration[]
+  >;
+  const [timeOptionsToDelete, setTimeOptionsToDelete] = useState<TimeOption[]>([]);
+  const [hasChangesNotificationConfig, setHasChangesNotificationConfig] = useState(false);
+
+  const setNotificationConfig = (val: NotificationConfiguration[]) => {
+    notificationConfig.current = val;
+    setHasChangesNotificationConfig(true);
+  };
+
+  const addTimeOptionsToDelete = (deletedElement: TimeOption) => {
+    // If it is not a newly created row that is yet to be persisted in the backend
+    if (deletedElement.id !== -1) {
+      const temp = [...timeOptionsToDelete];
+      temp.push(deletedElement);
+      setTimeOptionsToDelete(temp);
+    }
+  };
+
+  useEffect(() => {
+    dispatch(fetchNotificationConfigs());
+  }, [dispatch]);
+
+  useEffect(() => {
+    notificationConfig.current = cloneDeep(
+      session.notificationConfigs
+    ) as NotificationConfiguration[];
+  }, [session]);
+
+  const setIsEnabled = (index: number, value: boolean) => {
+    const temp = [...notificationConfig.current];
+    temp[index] = {
+      ...temp[index],
+      isEnabled: value
+    };
+    setNotificationConfig(temp);
+    gridApi.current?.getDisplayedRowAtIndex(index)?.setDataValue('isEnabled', value);
+    setHasChangesNotificationConfig(true);
+  };
+
+  const setTimeOptions = (index: number, value: TimeOption[]) => {
+    const temp = [...notificationConfig.current];
+
+    temp[index] = {
+      ...temp[index],
+      timeOptions: value
+    };
+    setNotificationConfig(temp);
+    gridApi.current?.getDisplayedRowAtIndex(index)?.setDataValue('timeOptions', value);
+    setHasChangesNotificationConfig(true);
+  };
+
+  const assessmentTypeFormatter: ValueFormatterFunc<NotificationConfiguration> = params => {
+    return params.data!.assessmentConfig?.type || '-';
+  };
+
+  const recipientFormatter: ValueFormatterFunc<NotificationConfiguration> = params => {
+    return params.data!.notificationType.forStaff ? 'Staff' : 'Student';
+  };
+
+  const columnDefs = [
+    {
+      headerName: 'Notification Type',
+      field: 'notificationType.name',
+      rowDrag: true
+    },
+    {
+      headerName: 'Assessment Type',
+      field: 'assessmentConfig.type',
+      valueFormatter: assessmentTypeFormatter
+    },
+    {
+      headerName: 'Recipients',
+      field: 'notificationType.forStaff',
+      valueFormatter: recipientFormatter
+    },
+    // {
+    //   headerName: 'View Email Template',
+    //   field: 'notificationType.name'
+    // },
+    // {
+    //   headerName: 'Past 30 Days',
+    //   field: 'notificationType.id'
+    // },
```

Comment:
Is this meant to be a future in-the-works extension? Could we leave a note here if that's the case?

Reply:
Yup, sure!